lib/msal-common/src/authority/AuthorityFactory.ts

```diff
@@ -20,13 +26,31 @@ export class AuthorityFactory {
         const authorityUrl = new UrlString(authorityString);
         const components = authorityUrl.getUrlComponents();
         const pathSegments = components.PathSegments;
-        switch (pathSegments[0]) {
-            case "tfp":
-                // tfp denotes a b2c url
-                return AuthorityType.B2C;
-            default:
-                // default authority is always AAD
-                return AuthorityType.Aad;
+
+        if (pathSegments[0] === "adfs")
+            return AuthorityType.Adfs;
+        else if (Object.keys(B2CTrustedHostList).length)
+            return AuthorityType.B2C;
+
+        // defaults to Aad
+        return AuthorityType.Aad;
+    }
+
+    /**
+     * @hidden
+     * @ignore
+     * Use when Authority is B2C is set to True to provide list of allowed domains.
+     * @param knownAuthorities
+     */
+    public static setKnownAuthorities(knownAuthorities: Array<string>): void {
+        if (!Object.keys(B2CTrustedHostList).length){
+            if (!knownAuthorities.length) {
+                throw ClientConfigurationError.createKnownAuthoritiesNotSetError();
```

Comment:
Need to remove this since we got rid of the authorityType parameter. This will throw for AAD use cases, which we dont want